dhis_android_analytics/src/main/java/dhis2/org/analytics/charts/ui/GroupAnalyticsFragment.kt

```diff
@@ -0,0 +1,138 @@
+package dhis2.org.analytics.charts.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import dhis2.org.R
+import dhis2.org.analytics.charts.data.AnalyticGroup
+import dhis2.org.analytics.charts.di.AnalyticsComponentProvider
+import dhis2.org.analytics.charts.ui.di.AnalyticsFragmentModule
+import dhis2.org.databinding.AnalyticsGroupBinding
+import dhis2.org.databinding.AnalyticsItemBinding
+import javax.inject.Inject
+import org.dhis2.commons.bindings.clipWithRoundedCorners
+
+const val ARG_MODE = "ARG_MODE"
+const val ARG_UID = "ARG_UID"
+
+class GroupAnalyticsFragment : Fragment() {
+
+    private val mode: AnalyticMode by lazy {
+        arguments?.getString(ARG_MODE)?.let { AnalyticMode.valueOf(it) } ?: AnalyticMode.HOME
+    }
+    private val uid: String? by lazy {
+        arguments?.getString(ARG_UID)
+    }
+    private lateinit var binding: AnalyticsGroupBinding
+
+    @Inject
+    lateinit var analyticsViewModelFactory: GroupAnalyticsViewModelFactory
+    private val groupViewModel: GroupAnalyticsViewModel by viewModels { analyticsViewModelFactory }
+    private val adapter: AnalyticsAdapter by lazy { AnalyticsAdapter() }
+
+    var disableToolbarElevation: (() -> Unit)? = null
+
+    companion object {
+        fun forEnrollment(enrollmentUid: String): GroupAnalyticsFragment {
+            return GroupAnalyticsFragment().apply {
+                arguments = bundleArguments(AnalyticMode.ENROLLMENT, enrollmentUid)
+            }
+        }
+
+        fun forProgram(programUid: String): GroupAnalyticsFragment {
+            return GroupAnalyticsFragment().apply {
+                arguments = bundleArguments(AnalyticMode.PROGRAM, programUid)
+            }
+        }
+
+        fun forHome(): GroupAnalyticsFragment {
+            return GroupAnalyticsFragment().apply {
+                arguments = bundleArguments(AnalyticMode.HOME)
+            }
+        }
+
+        fun forDataSet(dataSetUid: String): GroupAnalyticsFragment {
+            return GroupAnalyticsFragment().apply {
+                arguments = bundleArguments(AnalyticMode.DATASET, dataSetUid)
+            }
+        }
+    }
```

Comment:
Do you think is better to have only one method and pass the analytic mode as parameter?